worlds/civ_6/Locations.py

```diff
@@ -0,0 +1,153 @@
+from typing import List, Optional, Dict, Union
+from BaseClasses import Location, LocationProgressType, Region
+
+from .Data import CivicPrereqData, TechPrereqData, get_boosts_data, get_new_civics_data, get_new_techs_data
+
+from .Enum import CivVICheckType, EraType
+
+CIV_VI_AP_LOCATION_ID_BASE = 5041000
+
+# Locs that should not have progression items
+EXCLUDED_LOCATIONS = [
+    "GOODY_HUT_1",
+    "GOODY_HUT_2",
+    "GOODY_HUT_3",
+    "GOODY_HUT_4",
+    "GOODY_HUT_5",
+    "GOODY_HUT_6",
+    "GOODY_HUT_7",
+    "GOODY_HUT_8",
+    "GOODY_HUT_9",
+    "GOODY_HUT_10",
+]
+
+
+class CivVILocationData:
+    game: str = "Civilization VI"
+    name: str
+    cost: int
+    uiTreeRow: int
+    civ_id: int
+    code: int
+    era_type: str
+    location_type: CivVICheckType
+    pre_reqs: Optional[List[Union[CivicPrereqData, TechPrereqData]]]
+
+    def __init__(self, name: str, cost: int, uiTreeRow: int, id: int, era_type: str, location_type: CivVICheckType):
+        self.name = name
+        self.cost = cost
+        self.uiTreeRow = uiTreeRow
+        self.civ_id = id
+        self.code = id + CIV_VI_AP_LOCATION_ID_BASE
+        self.era_type = era_type
+        self.location_type = location_type
+
+
+class CivVILocation(Location):
+    game: str = "Civilization VI"
+    location_type: CivVICheckType
+
+    def __init__(self, player: int, name: str = "", address: Optional[int] = None, parent: Optional[Region] = None):
+        super().__init__(player, name, address, parent)
+        if name.split("_")[0] == "TECH":
+            self.location_type = CivVICheckType.TECH
+        elif name.split("_")[0] == "CIVIC":
+            self.location_type = CivVICheckType.CIVIC
+        elif name.split("_")[0] == "ERA":
+            self.location_type = CivVICheckType.ERA
+        elif name.split("_")[0] == "GOODY":
+            self.location_type = CivVICheckType.GOODY
+        elif name.split("_")[0] == "BOOST":
+            self.location_type = CivVICheckType.BOOST
+        else:
+            self.location_type = CivVICheckType.EVENT
+
+        if self.name in EXCLUDED_LOCATIONS:
+            self.progress_type = LocationProgressType.EXCLUDED
```

Comment:
I think it would make sense to move this to `generate_early` or somewhere else since it doesn't matter immediately and would take an `if` check out of every location creation in favor of just looping over the excluded locations and updating them later.

Reply:
Nice, that makes sense 👍

Reply:
Moved it to the end of `create_regions` since locations are not yet created in `generate_early` 👍